src/ripple/app/misc/SHAMapStoreImp.cpp

```diff
@@ -593,6 +584,58 @@ SHAMapStoreImp::clearPrior (LedgerIndex lastRotated)
     if (health())
         return;
 
+    {
+        auto const deleteBatch = setup_.deleteBatch;
+        auto const continueLimit = (deleteBatch + 1) / 2;
+
+        // This does make the sql harder to read
+        std::string const deleteQuery (
+            R"sql(
+            DELETE FROM Validations WHERE LedgerHash IN
```

Comment:
Mainly concerned about locking SQL, causing the peer to peer state to go non-full, then age getting high. DEBUG and WARN level messages say if the process aborts for some reason. I basically am concerned that the query scales O(n). Index searches each should be O(log n). Well, maybe I answered my own question. There's up to 3 index searches. So O(log 3n)? That's what the query planner in my head is saying.

So, it should be fine. If next month's online deletion of s1 hosts doesn't work, this will need to be revisited.

Let me know how this has been tested (not with unit tests but with real tests) please.